etcdserver/api/v3rpc/watch.go

```diff
@@ -76,6 +80,13 @@ func (sws *serverWatchStream) recvLoop() error {
 			return err
 		}
 
+		rh := &pb.ResponseHeader{
```

Comment:
how about defining NewHeader(rev int64)

And then do NewHeader(sws.watcherStream.Rev()) here?